xmppserver/src/main/java/org/jivesoftware/openfire/streammanagement/StreamManager.java

```diff
@@ -206,7 +222,11 @@ private void enable( String namespace, boolean resume )
         Element enabled = new DOMElement(QName.get("enabled", namespace));
         if (this.resume) {
             enabled.addAttribute("resume", "true");
-            enabled.addAttribute( "id", smId);
+            enabled.addAttribute("id", smId);
+            if ( !namespace.equals(NAMESPACE_V2) && LOCATION_ENABLED.getValue() ) {
+                // OF-1925: Hint clients to do resumes at the same cluster node.
+                enabled.addAttribute("location", XMPPServer.getInstance().getServerInfo().getHostname());
```

Comment:
(If this is a repeat, apologies, earlier comment along similar may have been lost)
https://xmpp.org/extensions/xep-0198.html#enable suggests that the location may optionally include a port. Might be worth including here too.

Reply:
PS. While you have the code open, is it worth adding the max="<max resumption time in seconds" attribute too.

Reply:
That's trickier to do, as you'd have to check if the current session was using Direct TLS or not. I'd rather not make this more complicated than needed.